backend/s3/s3.go

```diff
@@ -2638,12 +2638,6 @@ func NewFs(ctx context.Context, name, root string, m configmap.Mapper) (fs.Fs, e
 	if opt.Versions && opt.VersionAt.IsSet() {
 		return nil, errors.New("s3: cant use --s3-versions and --s3-version-at at the same time")
 	}
-	if opt.ACL == "" {
-		opt.ACL = "private"
-	}
-	if opt.BucketACL == "" {
-		opt.BucketACL = opt.ACL
```

Comment:
I think we still need these 3 lines otherwise we'll cause a backwards compatibility problem where users have set `acl` and are expecting `bucket_acl` to default to that.

```go
if opt.BucketACL == "" {
		opt.BucketACL = opt.ACL
}
```

Reply:
Rather than using the special case `unset` I've decided that empty string should mean not to supply the x-amz-acl header.

I'm going to write a bit more about this below (not in a review comment)!